### ISSUE TITLE
Consider usability of Validation Component

### ISSUE THREAD

**Issue by mahoneycm** [uswds/uswds#5526]
## Description

Validation findings, recommendations, and next steps from the [Aug-Sep 2023 Usability Testing Findings Report 🔒](https://docs.google.com/document/d/1z1BRxYZq_m65IdsRKsMzDYSpYYHKsqYgelQe7w_qDUQ/edit?usp=sharing)

## Findings report

Participants (4) were confused on what the purpose of this validation message was (we have seen 2 others in previous testing say the same thing (see below)), and no one noticed the state of it changing (the checkmark) once they entered the correct email. Users expect an error message (in-line validation or error at hitting submit button) if the input is invalid instead.

Previous research from patterns team:

- UAT [Findings presentation slide 26](https://docs.google.com/presentation/d/10KwPR77BF4Xrp2Goz0St-YGuBVojSf0Fdm6tn5qEEsQ/edit?usp=sharing)
- Clip: [Participant not used to seeing validation](https://drive.google.com/file/d/12N1k4uI6upysLoExm9f3BgO2mIYafj-1/view?usp=sharing);
- Clip: [another person was not sure what the point of using validation was “Just ask for it.](https://drive.google.com/file/d/1mWjcYo1kXLGtYHX8Tbwzi-ujNBU_YFEc/view?usp=sharing)” Said it doesn’t serve a purpose, Said it seems verbose
- [Error message research and recommendations for USWDS presentation](https://docs.google.com/presentation/d/1SmjPMxol6JNpMaUhzXtVvkNcYXmnRsP-y-NOozKNuA0/edit?usp=sharing)

### 💡 Validation recommendations:

- Guidance should emphasize giving helpful feedback (e.g. proper error states) to users at the point of need, like inline validation and error message upon form submission. We need more error/validation guidance in general (as stated in this [error messaging presentation](https://docs.google.com/presentation/d/1SmjPMxol6JNpMaUhzXtVvkNcYXmnRsP-y-NOozKNuA0/edit?usp=sharing)).

### Original recommendations

- Reevaluate the proper use cases and implementations for validation.
- Guidance should emphasize giving helpful feedback (e.g. proper error states) to users at the point of need, like inline validation and error message upon form submission. 
- We need more error/validation guidance in general (as stated in this [error messaging presentation](https://docs.google.com/presentation/d/1SmjPMxol6JNpMaUhzXtVvkNcYXmnRsP-y-NOozKNuA0/edit?usp=sharing)).

### Finding 1: Improve helpfulness for AT users

Could this be improved by moving the validation between the form and the button?

Consider password validation patterns:

![image](https://github.com/uswds/uswds/assets/25211650/2533e869-3269-437f-b944-b7a0d8810261)

Screen reader users would come across the validation _after_ entering the information. If the input doesn't meet validation criteria, they would be notified before reaching the button.

### Finding 2: Proper error state

Users expect an error message (in-line validation or error at hitting submit button) if the input is invalid instead.

## Discussion

- Proper usage is unclear: What is the use case for validation? Is it useful? Implementation guidance for validation is unclear. If there is a way it can and should be used, it is not clear from our guidance. 
- Validation status is not obvious, perceivable, or announced. It doesn't announce when the conditions are met. The way it is announcing to screener readers is unclear and not working now.
- If validation does not appear next to the input, it is not useful.
- Is validation a pattern? Should we approach it that way?
- Need to get a11y team member's input on accessibility issues for 1st 2 issues below. (POUR)

**Comment by brunerae** [comment]
Deprecation of this Component will be tracked in #6155
